backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -264,5 +264,236 @@ bool ValidateTableKeys::preorder(const IR::DpdkAsmProgram *p) {
     return false;
 }
 
+const IR::Expression* CopyPropagationAndElimination::getIrreplaceableExpr(cstring str,
+                                                                        bool allowConst) {
+    if (dontEliminate.count(str))
+        return nullptr;
+    if (!str.startsWith("m."))
+        return nullptr;
+    auto expr = replacementMap[str];
+    const IR::Expression* prev = nullptr;
+    while (expr != nullptr
+        && (!allowConst ? expr->is<IR::Member>() : true)
+        && str != expr->toString()
```

Comment:
For the P4 IR toString is lossy, it does not capture the entire expression.
Perhaps this is different for your part of the IR.

Reply:
you call expr->toString() many times, perhaps you can do it only once.

Reply:
yes, will clean it up.